@xen-orchestra/backups/_cleanVm.js

```diff
@@ -77,26 +77,32 @@ const mergeVhdChain = limitConcurrency(1)(async function mergeVhdChain(chain, {
 const noop = Function.prototype
 
 const listVhds = async (handler, vmDir) => {
-  const vhds = []
-  await asyncMap(
-    await handler.list(`${vmDir}/vdis`, {
-      prependDir: true,
-    }),
-    async jobDir =>
-      asyncMap(
-        await handler.list(jobDir, {
-          prependDir: true,
-        }),
-        async vdiDir =>
-          vhds.push(
-            ...(await handler.list(vdiDir, {
-              filter: isVhdFile,
-              prependDir: true,
-            }))
-          )
-      )
-  )
-  return vhds
+  try {
+    const vhds = []
+    await asyncMap(
+      await handler.list(`${vmDir}/vdis`, {
+        prependDir: true,
+      }),
+      async jobDir =>
+        asyncMap(
+          await handler.list(jobDir, {
+            prependDir: true,
+          }),
+          async vdiDir =>
+            vhds.push(
+              ...(await handler.list(vdiDir, {
+                filter: isVhdFile,
+                prependDir: true,
+              }))
+            )
+        )
+    )
+    return vhds
+  } catch (error) {
+    if (error?.code !== 'ENOENT') {
+      throw error
+    }
+  }
```

Comment:
To avoir catching unexpected errors, catch directly list calls and simply returns no entries :-)

Reply:
As it's a common case, maybe we should handle this directly in `AbstractHandler#list`, what do you think?

Reply:
Why not!